File: ohara-common/src/test/scala/com/island/ohara/util/TestVersionUtil.scala
package com.island.ohara.util

import java.io.{File, FileInputStream}
import java.util.Properties

import com.island.ohara.rule.SmallTest
import org.junit.Test
import org.scalatest.Matchers

class TestVersionUtil extends SmallTest with Matchers {
  @Test
  def testVersion(): Unit = {
    val modulePath = new File(".").getAbsolutePath
    val filePath = modulePath.substring(0, modulePath.indexOf("/ohara-common")) + "/gradle.properties"
    val props = new Properties
    val input = new FileInputStream(new File(filePath))
    try props.load(input)
    finally input.close()
    props.getProperty("version") shouldBe VersionUtil.VERSION
  }
}

File: ohara-common/src/test/java/com/island/ohara/common/util/TestVersionUtil.java
package com.island.ohara.common.util;

import com.island.ohara.common.rule.SmallTest;
import org.junit.Assert;
import org.junit.Test;

public class TestVersionUtil extends SmallTest {

  @Test
  public void allMembersShouldExist() {
    Assert.assertNotNull(VersionUtil.DATE);
    Assert.assertFalse(VersionUtil.DATE.isEmpty());
    Assert.assertNotNull(VersionUtil.REVISION);
    Assert.assertFalse(VersionUtil.REVISION.isEmpty());
    Assert.assertNotNull(VersionUtil.USER);
    Assert.assertFalse(VersionUtil.USER.isEmpty());
    Assert.assertNotNull(VersionUtil.VERSION);
    Assert.assertFalse(VersionUtil.VERSION.isEmpty());
  }
}
